contract_manager/src/chains.ts

```diff
@@ -447,12 +447,20 @@ export class EvmChain extends Chain {
       );
     }
 
-    const deployedContract = await deployTx.send({
-      from: signer.address,
-      gas,
-      gasPrice: gasPrice.toString(),
-    });
-    return deployedContract.options.address;
+    let deployedContract = undefined;
+    try {
+      deployedContract = await deployTx.send({
+        from: signer.address,
+        gas,
+        gasPrice: gasPrice.toString(),
+      });
+    } catch (e) {
+      // RPC errors often have useful information in the non-primary message field. Log the whole error
+      // to simplify identifying the problem.
+      console.log(`Error deploying contract: ${JSON.stringify(e)}`);
+      throw e;
+    }
+    return deployedContract!.options.address;
```

Comment:
you can move this return inside the try and remove the non-null assertion + remove `let deployedContract = undefined;`